src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -382,6 +383,31 @@ private HttpMessageInvoker CreateInternalHttpInvoker(HttpMessageHandler? handler
         {
             handler = HttpHandlerFactory.CreatePrimaryHandler();
         }
+        else
+        {
+            // Validate the user specified handler is compatible with this platform.
+            //
+            // Android's native handler doesn't fully support HTTP/2 and using it could cause hard to understand errors
```

Comment:
Does this mean the native handler works for some cases? And what if the native handler improves in the future to have better support? Should there be a switch to ignore this check and allow using the native handler?

Reply:
My thought is if that happens, the grpc-dotnet client can be updated to detect the Android OS version that fully supports gRPC, and not throw this error.